### ISSUE TITLE
First Epoch Loss is Bigger than Expected

### ISSUE THREAD

**Issue by dg520** [davek44/Basset#41]
Hi David,

Thanks for developing the powerful Basset. I've tried to apply it on my splicing motif study and the result is quite positive. However, I have a little confusion about the math behind: 

My train and valid loss at the first epoch is about 1.9. As I've three classes in this study and I think `binary cross entropy` is used as the default loss function, this should give me a starting loss no more than `ln(3)`, which is around 1.1. So I don't know why all my models started at 1.9, which is way higher than the maximal expectation 1.1. 

Did I miss something or misunderstand some concepts? Thank you very much in advance.

Best,
Dadi

**Comment by davek44** [comment]
Hey Dadi, I'm a bit too tied up right now to dig in here myself. In Basset, the loss is defined here: https://github.com/davek44/Basset/blob/master/src/convnet.lua#L251. The binary cross entropy loss is described here: https://github.com/torch/nn/blob/master/doc/criterion.md#nn.BCECriterion. I'd open a torch session, and try some things to test your assumptions and get a feel for how these work.

**Comment by dg520** [comment]
Hi David,

Sorry my assumption was totally wrong. By generating a dummy set with random targets labels either 0 or 1 and normally distributed probability between 0.0001 and 0.99, the defined BCE loss function `loss(o, t) = - 1/n sum_i (t[i] * log(o[i]) + (1 - t[i]) * log(1 - o[i]))` can exceed my expectation. It makes sense now.

BTW, seems the nucleotide order in motif heat map is reversed. I've compared the weight matrix and the heat map matrix. both agree to the reverse of current setting. I've reported it in a separate thread. 

Apologize for disturbing you a lot. 

Best,
Dadi
